src/features/common/components/wallet-provider.tsx

```diff
@@ -24,9 +38,29 @@ export function WalletProvider({ networkConfig, children }: Props) {
               baseServer: networkConfig.kmd.server,
               token: networkConfig.kmd.token ?? '',
               port: String(networkConfig.kmd.port),
+              promptForPassword: async () => {
+                const password = await openKmdPasswordDialog({ message: 'Enter KMD Password' })
+                if (password === undefined || password == null) {
```

Comment:
```suggestion
                if (password == null) {
```
That covers the undefined scenario as well.